src/components/popup/SignUpPopup.js

```diff
@@ -0,0 +1,163 @@
+import PopupWithForm from './PopupWithForm';
+import React from 'react';
+
+export default function SignUpPopup(props) {
+  const { isOpen, onClose, handleSwitchPopup, handleSignup, buttonText, onPopupClick } = props;
+  const [isValid, setIsValid] = React.useState(false);
+  const [username, setUsername] = React.useState('');
+  const [password, setPassword] = React.useState('');
+  const [email, setEmail] = React.useState('');
+  const [isEmailCorrect, setIsEmailCorrect] = React.useState(true);
+  const [isPasswordCorrect, setIsPasswordCorrect] = React.useState(true);
+  const [shouldAddSSign, setShouldAddSSign] = React.useState(false);
+  const [passwordErrorText, setPasswordErrorText] = React.useState('Password incorrect');
+  const [isUsernameCorrect, setisUsernameCorrect] = React.useState(true);
+
+  // ! Reseting the popup when closing
+  React.useEffect(() => {
+    setIsEmailCorrect(true);
+    setIsPasswordCorrect(true);
+    setisUsernameCorrect(true);
+    setEmail('');
+    setPassword('');
+    setUsername('');
+    setIsValid(false);
+    setShouldAddSSign(false);
+  }, [isOpen]);
+
+  // ! Validating the email input
+  const checkEmailValid = () => {
+    // eslint-disable-next-line
+    const emailRegExp = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
+    if (emailRegExp.test(email)) {
+      if (email.length >= 8) {
+        setIsEmailCorrect(true);
+      } else {
+        setIsEmailCorrect(false);
+      }
+    } else {
+      if (email === '') {
+        setIsEmailCorrect(true);
+      } else {
+        setIsEmailCorrect(false);
+      }
+    }
+  };
+
+  // ! Validating the password input
+  const checkPasswordValid = () => {
+    const passwordRegExp = /^(?=.*[0-9])[a-zA-Z0-9!@#$%^&*]{6,16}$/;
+    const passwordSpecialSignRegExp = /(?=.*[!@#$%^&*])/;
+    if (password.length >= 6 || password === '') {
+      if (passwordRegExp.test(password)) {
+        if (!passwordSpecialSignRegExp.test(password)) {
+          setPasswordErrorText('It`s better to add a special sign ( ! @ # $ % ^ & * ).');
+          setShouldAddSSign(true);
+        } else {
+          setShouldAddSSign(false);
+          setPasswordErrorText('Password incorrect. Must contain numbers and letters!');
+        }
+        setIsPasswordCorrect(true);
+      } else {
+        setShouldAddSSign(false);
+        setPasswordErrorText('Password incorrect. Must contain numbers and letters!');
+        if (password === '') {
+          setIsPasswordCorrect(true);
+        } else {
+          setIsPasswordCorrect(false);
+        }
+      }
+    } else{
+      setIsPasswordCorrect(false);
+      setPasswordErrorText('Password to short!');
+    }
+  };
+
+  // ! Validating the name input
+  const checkUsernameValid = () => {
+    var usernameRegExp = /^[a-zA-Z0-9 ]{2,40}$/;
```

Comment:
please, don't use `var` in the project because it has some vulnerability. Only `let` and `const` are allowed